crates/optimism/chainspec/src/lib.rs

```diff
@@ -39,6 +47,86 @@ pub fn optimism_deposit_tx_signature() -> Signature {
     Signature::new(U256::ZERO, U256::ZERO, Parity::Parity(false))
 }
 
+impl EthChainSpec for OpChainSpec {
+    fn chain(&self) -> alloy_chains::Chain {
+        self.inner.chain()
+    }
+
+    fn base_fee_params_at_timestamp(&self, timestamp: u64) -> BaseFeeParams {
+        self.inner.base_fee_params_at_timestamp(timestamp)
+    }
+
+    fn base_fee_params_at_block(&self, block_number: u64) -> BaseFeeParams {
+        self.inner.base_fee_params_at_block(block_number)
+    }
+
+    fn deposit_contract(&self) -> Option<&DepositContract> {
+        self.inner.deposit_contract()
+    }
```

Comment:
followup here would be to check if we actually need this in the trait because this is just an ethereum setting.

it might be useful to split `EthChainSpec` into two traits

Reply:
this is only used for pruning configuration https://github.com/paradigmxyz/reth/blob/48a4702f3eac72d1aab6d4095caa331ffdde5dfa/crates/node/core/src/args/pruning.rs#L106-L111

I think we can either move `prune_config` fn to chainspec trait or introduce trait(s) which would be able to produce a concrete `*Config` type based on concrete cli `*Args` type and a chainspec

we also have similar pattern in `NetworkArgs` https://github.com/paradigmxyz/reth/blob/48a4702f3eac72d1aab6d4095caa331ffdde5dfa/crates/node/core/src/args/network.rs#L199

Reply:
I see, I guess this doesn't matter really if this is part of the trait since option